### PR TITLE
Follow symlinks while hard-linking migrated files.

### PR DESCRIPTION
**Pull request by kyrofa**
Currently Snapcraft does its best to migrate files efficiently by utilizing hard links. However, `os.link()` doesn't seem to respect its `follow_symlinks` parameter, which results in broken symlinks in the final snap. This PR fixes LP: [#1572664](https://bugs.launchpad.net/snapcraft/+bug/1572664), working around that issue by utilizing `os.path.realpath()`.

### CODE REVIEW

File: snapcraft/internal/common.py [hunk 1]
```diff
@@ -149,10 +149,16 @@ def reset_env():
...
+    # plan we'll just copy it.
     try:
-        os.link(source, destination, follow_symlinks=follow_symlinks)
+        # Note that follow_symlinks doesn't seem to work for os.link, so we'll
+        # implement this logic ourselves using realpath.
+        source_path = source
+        if follow_symlinks:
+            source_path = os.path.realpath(source)
+
+        os.link(source_path, destination, follow_symlinks=False)
```
Comment (sergiusens) [inline snapcraft/internal/common.py]: the `follow_symlinks` here is now irrelevant, right?

Comment (kyrofa) [inline snapcraft/internal/common.py]: Yeah but it defaults to True and I want to make sure this continues working even if the bug is fixed upstream.

Comment (sergiusens) [inline snapcraft/internal/common.py]: A code comment would come in handy then :-)

Comment (kyrofa) [inline snapcraft/internal/common.py]: Done :smiley: .

**Pull request merged**